x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DeleteStep.java

```diff
@@ -52,7 +53,10 @@ public void performDuringNoSnapshot(IndexMetadata indexMetadata, ClusterState cu
                 // This is the last backing index in the data stream, and it's being deleted because the policy doesn't have a rollover
                 // phase. The entire stream needs to be deleted, because we can't have an empty list of data stream backing indices.
                 // We do this even if there are multiple failure store indices because otherwise we would never delete the index.
-                DeleteDataStreamAction.Request deleteReq = new DeleteDataStreamAction.Request(new String[] { dataStream.getName() });
+                DeleteDataStreamAction.Request deleteReq = new DeleteDataStreamAction.Request(
+                    MasterNodeRequest.infiniteMasterNodeTimeout(currentState.getMinTransportVersion()),
```

Comment:
Slight behaviour change, previously we'd time out after 30s here, but waiting forever seems more appropriate.